### ISSUE TITLE
There is no 'esperanto' in the Settings/Language list

### ISSUE THREAD

**Issue by sikmir** [OpenOrienteering/mapper#792]
### Steps to reproduce
Esperanto translation was registered by commit 786a807995d136a68715cc0744cece742a79810f.
But still there is no 'esperanto' in the Settings/Language list.
It looks like Qt-related problem, the root cause below.

### Actual behaviour
`QLocale(QString::fromLatin1("eo")).language() == QLocale::C`

### Expected behaviour
`QLocale(QString::fromLatin1("eo")).language() == QLocale::Esperanto`

### Configuration
Mapper Version: latest master
Operating System: Linux, Qt 5.7.0.

**Comment by dg0yt** [comment]
Hm, we leverage QLocale. QLocale documentation says that it is to be initialized with a language/country pair (+ optional script). We don't have a (default) country in `"eo"`, and that's probably the reason why even the [`QLocale(const QString&)` constructor](http://doc.qt.io/qt-5/qlocale.html#QLocale-1) doesn't create a QLocale for the Esperanto language.

But even an explicit `(QLocale(QLocale::Esperanto, QLocale::AnyScript, QLocale::AnyCountry)` or similar doesn't create a QLocale with the Esperanto language in Qt 5.6.1.

Related QTBUGs:
https://bugreports.qt.io/browse/QTBUG-14592?jql=text%20~%20%22esperanto%22

**Comment by sikmir** [comment]
[QTBUG-14592](https://bugreports.qt.io/browse/QTBUG-14592) was closed as Done (Qt fix version 5.2.0). But it depends on [QTBUG-3701](https://bugreports.qt.io/browse/QTBUG-3701) and there is no activity there since 2013. So it seems there is nothing to wait for fix in the near future. All we can do is use a workaround in one form or another.

**Comment by dg0yt** [comment]
QTBUG-14592 is just about how the issue used to show up in Qt Linguist, and it was solved using private API which I don't want to use. QTBUG-3701 could help to mitigate but not solve our issue. We just cannot use QLocale as convenient as we did so far.

What I find disappointing is the fact that there is no mention of such limitations in Qt documentation. My understanding is that something like 

~~~
QCOMPARE(QLocale(QLocale::Esperanto, QLocale::AnyScript, QLocale::AnyCountry).language(), QLocale::Esperanto);
~~~

should work at least, but it is not the case. I wonder if I should to open a QTBUG.
